pkg/odo/cli/url/create.go

```diff
@@ -204,52 +210,62 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 
 // Validate validates the URLCreateOptions based on completed values
 func (o *URLCreateOptions) Validate() (err error) {
-	// Check if exist
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
+	if !util.CheckOutputFlag(o.OutputFlag) {
+		return fmt.Errorf("given output format %s is not supported", o.OutputFlag)
+	}
+
+	// if experimental mode is enabled, and devfile is provided.
+	errorList := make([]string, 0)
```

Comment:
Since we are now collecting the validation issues and displaying at once, we should also include the tls secret failure with non-secure URLs.
```
[mrinaldas@localhost nodejs-ex]$ odo url create node --tls-secret djsdk
 ✗  tls secret is only available for secure URLs of ingress kind
[mrinaldas@localhost nodejs-ex]$ odo url create node --host com
 ✗  There are validation issues:
        - host is not supported for URLs of Route Kind
        - URL node already exists
```

Reply:
Good point.